src/System.Management.Automation/utils/Telemetry.cs

```diff
@@ -0,0 +1,403 @@
+                s_telemetryClient = new TelemetryClient();
+                TelemetryConfiguration.Active.InstrumentationKey = _psCoreTelemetryKey;
+                // Set this to true to reduce latency during development
+                TelemetryConfiguration.Active.TelemetryChannel.DeveloperMode = false;
+                s_sessionId = Guid.NewGuid().ToString();
+
+                // use a hashset when looking for module names, it should be quicker than a string comparison
+                s_knownModules = new HashSet<string>(
+                    new string []
+                    {
+                        "Microsoft.PowerShell.Archive",
+                        "Microsoft.PowerShell.Host",
+                        "Microsoft.PowerShell.Management",
+                        "Microsoft.PowerShell.Security",
+                        "Microsoft.PowerShell.Utility",
+                        "PackageManagement",
+                        "Pester",
+                        "PowerShellGet",
+                        "PSDesiredStateConfiguration",
+                        "PSReadLine",
+                        "ThreadJob",
+                    },
+                    StringComparer.OrdinalIgnoreCase
+                    );
+                s_uniqueUserIdentifier = GetUniqueIdentifier().ToString();
+            }
+        }
+
+        /// <summary>
+        /// Determine whether the environment variable is set and how.
+        /// </summary>
+        /// <param name="name">The name of the environment variable.</param>
+        /// <param name="defaultValue">If the environment variable is not set, use this as the default value.</param>
+        /// <returns>A boolean representing the value of the environment variable.</returns>
+        private static bool GetEnvironmentVariableAsBool(string name, bool defaultValue)
+        {
+            var str = Environment.GetEnvironmentVariable(name);
+            if (string.IsNullOrEmpty(str))
+            {
+                return defaultValue;
+            }
+
+            switch (str.ToLowerInvariant())
+            {
+                case "true":
+                case "1":
+                case "yes":
+                    return true;
+                case "false":
+                case "0":
+                case "no":
+                    return false;
+                default:
+                    return defaultValue;
+            }
+        }
+
+        /// <summary>
+        /// Send telemetry as a metric.
+        /// </summary>
+        /// <param name="metricId">The type of telemetry that we'll be sending.</param>
+        /// <param name="data">The specific details about the telemetry.</param>
+        internal static void SendTelemetryMetric(TelemetryType metricId, string data)
+        {
+            if (!CanSendTelemetry)
+            {
+                return;
+            }
+
+            string metricName = metricId.ToString();
+            string uuidString = s_uniqueUserIdentifier;
```

Comment:
It seems we can use `s_uniqueUserIdentifier` in the `TrackValue` calls directly without needing `uuidString`.

Reply:
fixed